scripts/platform-init/platform-init.py

```diff
@@ -86,20 +102,24 @@ def init_secret_service_secret(config: Config):
                 type="Opaque",
             ),
         )
-    elif existing_private_key is None and config.secret_service_private_key is not None:
+    elif config.secret_service_private_key is not None:
         # create private key from config
+        logging.info("Setting private key from values")
+        if existing_private_key:
+            # delete old private key
+            logging.info("deleting old private key secret")
+            v1.delete_namespaced_secret(private_key_secret, config.k8s_namespace)
```

Comment:
suggestion(blocking): Just fail here. Because it is very likely someone is making a mistake in this case. And the key that is being deleted is still in use. The way I understand this branch of the logic is that there is an existing private key secret in k8s and also a secret has been set in the configuration but the previous secret field is not set. So in this case most likely there are secrets in the DB encrypted with the value in the k8s secret. In this case the only viable solution is to do a rotation - otherwise we may get in trouble.

Reply:
This happens when you rotate the secret, you copy the key to previousKey and set a new key in key. I'll change it so it only does this if a previousKey was set.